modules/dhcp_native_ms/dhcp_native_ms_main.rb

```diff
@@ -78,6 +78,41 @@ def unused_ip(subnet_address, mac_address, from_address, to_address)
         raise Proxy::DHCP::NotImplemented.new("DhcpsApi::Server#get_free_ip_address is not available on Windows Server 2008 and earlier versions.")
       end
 
+      dhcp_range = dhcpsapi.list_subnet_elements(subnet_address, DhcpsApi::DHCP_SUBNET_ELEMENT_TYPE::DhcpIpRanges)
+      dhcp_start_addr = dhcp_range.map {|r| r[:element][:start_address]}.first
+      dhcp_end_addr = dhcp_range.map {|r| r[:element][:end_address]}.first
+
+      if from_address.to_s.empty?
```

Comment:
from_address and to_address will be nils if the caller skipped these parameters. ```from_address.nil?``` is better check in this case.

Reply:
Right, I will update it.